x/fungible/keeper/gas_coin_and_pool.go

```diff
@@ -63,7 +63,7 @@ func (k Keeper) setupChainGasCoinAndPool(ctx sdk.Context, chainID int64, gasAsse
 	if err != nil {
 		return ethcommon.Address{}, sdkerrors.Wrapf(err, "failed to get system contract abi")
 	}
-	_, err = k.CallEVM(ctx, *systemABI, types.ModuleAddressEVM, systemContractAddress, BigIntZero, nil, true, "setGasZetaPool", big.NewInt(chain.ChainId), zrc20Addr)
+	_, err = k.CallEVM(ctx, *systemABI, types.ModuleAddressEVM, systemContractAddress, BigIntZero, nil, true, false, "setGasZetaPool", big.NewInt(chain.ChainId), zrc20Addr)
```

Comment:
I'm curious that we had a lots of `CallEVM` calls setting `commit` flag as `true`. Maybe they were all old mistakes?

Reply:
I didn't change the value for `commit` in this PR, I introduced a new argument after it that allows us to prevent eventually emitting Ethereum events because it creates a bug in some cases.
`commit` is necessary to update the state which is the case here

Reply:
Gotta.